File: engine/transpile/src/test/resources/golden/SYK004/WS_作業項目.py
"""SYK004 のデータ部レコード WS-作業項目 を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_作業項目:
    """01 WS-作業項目 総バイト長 9。"""

    LENGTH = 9

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(9)
        else:
            self._data = bytearray(data)
            if len(self._data) < 9:
                self._data.extend(b"\x00" * (9 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-在庫残数
    def get_WS_在庫残数(self):
        return decode_packed(self._data, 0, 4)
    def set_WS_在庫残数(self, value):
        encode_packed(self._data, 0, 4, True, value)

    # WS-引当数量
    def get_WS_引当数量(self):
        return decode_packed(self._data, 4, 4)
    def set_WS_引当数量(self, value):
        encode_packed(self._data, 4, 4, True, value)

    # WS-判定区分
    def get_WS_判定区分(self):
        return decode_alphanumeric(self._data, 8, 1)
    def set_WS_判定区分(self, value):
        encode_alphanumeric(self._data, 8, 1, value)

    def is_WS_在庫あり(self):
        return self.get_WS_判定区分() == "1"

    def is_WS_在庫なし(self):
        return self.get_WS_判定区分() == "2"

File: engine/transpile/src/test/resources/golden/SYK004/WS_作業項目.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK004 のデータ部レコード WS-作業項目 を逐語対訳した自動生成コード。 */
public final class WS_作業項目 {

    /** 01 WS-作業項目 総バイト長 9。 */
    public static final int LENGTH = 9;

    private final byte[] data;

    public WS_作業項目() {
        this.data = new byte[9];
    }

    public WS_作業項目(byte[] source) {
        this.data = Arrays.copyOf(source, 9);
    }

    public byte[] data() {
        return data;
    }

    // WS-在庫残数
    public long get_WS_在庫残数() {
        return CobolRuntime.decodePacked(data, 0, 4);
    }
    public void set_WS_在庫残数(long value) {
        CobolRuntime.encodePacked(data, 0, 4, true, value);
    }

    // WS-引当数量
    public long get_WS_引当数量() {
        return CobolRuntime.decodePacked(data, 4, 4);
    }
    public void set_WS_引当数量(long value) {
        CobolRuntime.encodePacked(data, 4, 4, true, value);
    }

    // WS-判定区分
    public String get_WS_判定区分() {
        return CobolRuntime.decodeAlphanumeric(data, 8, 1);
    }
    public void set_WS_判定区分(String value) {
        CobolRuntime.encodeAlphanumeric(data, 8, 1, value);
    }

    public boolean is_WS_在庫あり() {
        return get_WS_判定区分().equals("1");
    }

    public boolean is_WS_在庫なし() {
        return get_WS_判定区分().equals("2");
    }
}
